src/snowflake/snowpark/session.py

```diff
@@ -2749,4 +2750,34 @@ def _explain_query(self, query: str) -> Optional[str]:
             _logger.warning("query `%s` cannot be explained", query)
             return None
 
+    @staticmethod
+    def connect() -> "Session":
+        """Gets the existing session if one exists (in case of SPs/Streamlits) or creates a new one using default connection params (e.g., in case of SPCS)."""
+
+        # 1) check if in SiS or sproc
+        if (
+            is_in_stored_procedure()
+        ):  # (will be removed before merging) no need for `"streamlit" in sys.modules` check since is_in_stored_procedure() must be true for SiS as well
+            # Basically noop and return the existing session
```

Comment:
it would be removed before merging to main? I don't understand why have it then.

Reply:
There is already `Session.builder.getOrCreate()`

Reply:
+1

Reply:
`getOrCreate` as is, doesn't work (since we expect to have token in the toml file and not as a separate file). But I had a conversion with Srikanth, he is tasked with adding `connect` for SPCS for _all_ connectors. He will write a design doc based on the existing `connect` API in Python connector where we take into account a new connect argument (like token_file) to be added to the toml file. I'll hold off working on this for now since his general solution is better/cleaner than what I have here.